botx/models/smartapps.py

```diff
@@ -0,0 +1,117 @@
+"""Definition of smartapp object."""
+
+from typing import Any, BinaryIO, Dict, List, Optional, TextIO, Union
+from uuid import UUID
+
+from botx.models.base import BotXBaseModel
+from botx.models.files import SmartAppFile
+from botx.models.messages.message import Message
+
+
+class SendingSmartAppEvent(BotXBaseModel):
+    """SmartApp event with data."""
+
+    #: uniq id of request
+    ref: Optional[UUID] = None
+
+    #: id of smartapp
+    smartapp_id: UUID
+
+    #: payload of smartapp
+    data: Union[Dict, List]  # noqa: WPS110
+
+    #: options of smartapp
+    opts: Dict[str, Any] = {}
+
+    #: version of protocol smartapp <-> bot
+    smartapp_api_version: int
+
+    #: chat of this smartapp (smartapp_id now)
+    group_chat_id: UUID
+
+    # smartapp files
+    files: List[SmartAppFile] = []
+
+    @classmethod
+    def from_message(
+        cls,
+        data: Union[Dict, List],  # noqa: WPS110
+        message: Message,
+    ) -> "SendingSmartAppEvent":
+        """Build smartapp event from message.
+
+        Arguments:
+            data: smartapp's data.
+            message: incoming message.
+
+        Returns:
+            Built smartapp event.
+        """
+        return cls(
+            ref=message.data["ref"],
+            smartapp_id=message.data["smartapp_id"],
+            data=data,
+            opts=message.data["opts"],
+            smartapp_api_version=message.data["smartapp_api_version"],
+            group_chat_id=message.group_chat_id,
+        )
+
+    def add_file(
+        self,
+        file: Union[TextIO, BinaryIO, SmartAppFile],
+        filename: Optional[str] = None,
+        file_id: Optional[UUID] = None,
+    ) -> None:
+        """Attach file to smartapp.
+
+        Arguments:
+            file: file that should be attached to the message.
+            filename: name for file that will be used if if can not be retrieved from
+                file.
+            file_id: unique file identifier.
+        """
+        if isinstance(file, SmartAppFile):
+            file.file_name = filename or file.file_name
```

Comment:
Mb raise ValueError if file is SmartAppFile and file_id is not None?

Reply:
We have decided to remove `file_id`.